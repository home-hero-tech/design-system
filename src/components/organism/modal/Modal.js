import React from 'react';
import PropTypes from 'prop-types';
import ReactModal from 'react-modal';
import classNames from 'classnames';

import css from './Modal.module.scss';

import CloseButton from './CloseButton';

if (process.env.NODE_ENV !== 'test') ReactModal.setAppElement('#root');

const Modal = ({
  open,
  children,
  left,
  right,
  center,
  onRequestClose,
  contentLabel,
  widthLevel,
  hasPadding,
  onCancel,
  className,
  ...otherProps
}) => {
  const classes = classNames(css['c-modal'], className, {
    [css['c-modal--pre-left']]: left,
    [css['c-modal--pre-right']]: right,
    [css[`c-modal--width-${widthLevel}`]]: true,
    [css['c-modal--center']]: center,
    [css['c-modal--has-padding']]: hasPadding,

    [css['c-modal-overlay--left']]: left,
    [css['c-modal-overlay--right']]: right
  });

  const classesOverlay = classNames({
    [css['c-modal-overlay']]: true,
    [css['c-modal-overlay--left']]: left,
    [css['c-modal-overlay--right']]: right
  });

  return (
    <ReactModal
      isOpen={open}
      className={classes}
      overlayClassName={classesOverlay}
      contentLabel={contentLabel}
      shouldCloseOnOverlayClick
      closeTimeoutMS={300}
      onRequestClose={onRequestClose}
      ariaHideApp={process.env.NODE_ENV !== 'test'}
      {...otherProps}
    >
      {onCancel ? <CloseButton onClick={onCancel} /> : null}
      {children}
    </ReactModal>
  );
};

Modal.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.object,
    PropTypes.array,
    PropTypes.string
  ]).isRequired,
  contentLabel: PropTypes.string.isRequired,
  left: PropTypes.bool,
  onRequestClose: PropTypes.func,
  onCancel: PropTypes.func,
  open: PropTypes.bool.isRequired,
  center: PropTypes.bool,
  right: PropTypes.bool,
  hasPadding: PropTypes.bool,
  widthLevel: PropTypes.number,
  className: PropTypes.string
};

Modal.defaultProps = {
  left: false,
  right: true,
  hasPadding: true,
  onRequestClose: null,
  onCancel: null,
  widthLevel: 1,
  center: false,
  className: ''
};

export default Modal;
