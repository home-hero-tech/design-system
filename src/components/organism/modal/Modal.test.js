import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Modal from './Modal';

describe('Modal', () => {
  it('should match with the snapshot', () => {
    const isOpen = true;

    const wrapper = shallow(
      <div id="root">
        <Modal
          open={isOpen}
          contentLabel="Modal"
          onConfirm={() => console.log('confirm')}
          onCancel={() => console.log('cancel')}
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Odio, vitae!
        </Modal>
      </div>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
