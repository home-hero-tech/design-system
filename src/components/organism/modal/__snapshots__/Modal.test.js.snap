// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should match with the *left* props 1`] = `
<div
  className="root"
  id="root"
>
  <Modal
    center={false}
    className=""
    contentLabel="Default Modal"
    hasPadding={true}
    left={true}
    onCancel={null}
    onRequestClose={null}
    open={true}
    right={true}
    widthLevel={1}
  >
    Lorem ipsum dolor sit amet, consectetur adipisicing elit. Odio, vitae!
  </Modal>
</div>
`;

exports[`Modal should match with the *right* props 1`] = `
<div
  className="root"
  id="root"
>
  <Modal
    center={false}
    className=""
    contentLabel="Default Modal"
    hasPadding={true}
    left={false}
    onCancel={null}
    onRequestClose={null}
    open={true}
    right={true}
    widthLevel={1}
  >
    Lorem ipsum dolor sit amet, consectetur adipisicing elit. Odio, vitae!
  </Modal>
</div>
`;

exports[`Modal should match with the snapshot 1`] = `
<div
  className="root"
  id="root"
>
  <Modal
    center={false}
    className=""
    contentLabel="Default Modal"
    hasPadding={true}
    left={false}
    onCancel={null}
    onRequestClose={null}
    open={true}
    right={true}
    widthLevel={1}
  >
    Lorem ipsum dolor sit amet, consectetur adipisicing elit. Odio, vitae!
  </Modal>
</div>
`;
